### DISCUSSION TITLE
Create sticky session via kong (without kong ingress) on k8s

### DISCUSSION THREAD

**Discussion by shahargAnv** [Kong/kong#7221] [Help] [answered]
Hi everyone, 
i have k8s cluster
using nginx-ingress and kong for api-gateway . 

i'm using a socket.io svc that requires using sticky session (client have to request the same pod everytime) 

i'm trying to use the k8s 
`  sessionAffinity: ClientIP `

but everytime i'm getting into the same pod, because the request always goes with the same ip (the kong pod ip) 

how can i overcome this problem?

**Comment by mayocream** [comment, marked as answer]
Summary:  Traffic --> ingress-nginx -->  Kong Pod --> (svc) Application (Sticky session)

So there are three different settings for your needs.

1. [ingress-nginx sticky session annotations](https://kubernetes.github.io/ingress-nginx/examples/affinity/cookie/#sticky-sessions)
2. [Kong upstream load balancing algorithms](https://docs.konghq.com/gateway/2.7.x/reference/loadbalancing/#balancing-algorithms). Note that if you are using Kong with K8s svc, which should be [DNS-based load balancing](https://docs.konghq.com/gateway/2.7.x/reference/loadbalancing/#dns-based-load-balancing), it depends on your CoreDNS (or other DNS server) in your K8s cluster.
3. [K8s service spec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#servicespec-v1-core) - sessionAffinity

You can go through these settings to ensure they are correct.
